packages/server/src/adapters/node-http/incomingMessageToRequest.ts

```diff
@@ -0,0 +1,94 @@
+import type * as http from 'http';
+import { TRPCError } from '../../@trpc/server';
+
+export interface IncomingMessageWithBody extends http.IncomingMessage {
+  /**
+   * Many adapters will add a `body` property to the incoming message and pre-parse the body
+   */
+  body?: unknown;
+}
+/**
+ * Convert an incoming message to a body stream with a max size
+ */
+function incomingMessageToBodyStream(
+  req: IncomingMessageWithBody,
+  opts: { maxBodySize: number | null },
+) {
```

Comment:
This function only exists to allow for max body size, otherwise we could just send `req` as the body as event emitters are streams

Should we ditch `maxBodySize` altogether perhaps?